prost-build/src/lib.rs

```diff
@@ -144,7 +144,7 @@ type Module = Vec<String>;
 /// implementation of the generated trait in the application code and plugs it into the framework.
 ///
 /// Such framework isn't part of Prost at present.
-pub trait ServiceGenerator {
+pub trait ServiceGenerator: std::fmt::Debug {
```

Comment:
This is a breaking change, so I'd prefer not to include it.

Reply:
What does it break?  I mean, I can take it out if need be, but I really don't think it can break anything.

Reply:
If this were released, existing `ServiceGenerator` impls which don't impl `Debug` would fail to compile, thus it's a breaking change.  Yes, please take it out.

Reply:
Fair enough, would you be open to a manual implementation that doesn't have that problem?

Reply:
What is a manual implementation?

Reply:
`impl Debug for Config { ... }`

Reply:
The breaking change comment applies only to the highlighted line, which adds a trait superbound.  Adding `#[derive(Debug)]` to `Config` is fine.

Reply:
It can't be derived because deriving Debug requires the breaking change you're talking about, that's why I'm suggesting doing it manually.

Reply:
Adding a trait superbound and adding a Debug impl for a struct are two different things.  My comments in this thread only apply to the former.

Reply:
ah ok I see, yeah you can add a manual Debug impl for Config, if the derive impl is assuming that supertrait bound.

Reply:
(I thought it just added a where clause to the impl though?)

Reply:
I don't think it's quite that easy because it's using `dyn` dispatch